src/awx_plugins/credentials/plugins.py

```diff
@@ -744,10 +744,29 @@
                 'type': 'boolean',
                 'secret': False,
             },
+            {
+                'id': 'request_timeout',
+                'label': gettext_noop('Verify SSL'),
+                'type': 'string',
+                'secret': False,
+                'default': 'galaxy',
+                'help_text': gettext_noop(
+                    'Specify the timeout Ansible should use in requests to'
+                    'the host. Defaults to 10s',
+                ),
+            },
         ],
         'required': ['host'],
     },
     injectors={
+        'extra_vars': {
+            'aap_hostname': '{{host}}',
+            'aap_username': '{{username}}',
+            'aap_password': '{{password}}',
+            'aap_token': '{{oauth_token}}',
+            'aap_request_timeout': '{{request_timeout}}',
+            'aap_validate_certs': '{{verify_ssl}}',
+        },
```

Comment:
I'm only slightly worried that adding extra_vars may introduce compatibility issues, because some credential uses don't take extra_vars, like inventory imports. But I haven't been involved much with awx-plugins and I'd like someone who has a little more recent experience to comment.

Reply:
@AlanCoding I don't think this will be picked up for inventories, and its not built to do that, This was added specifically to work with the aap_configuration collection, and if the user was wanting to use it in a playbook, Have no clue if it would work with inventories, how does it work now(I dont think it works as is with extra vars, or at least defaults to the env vars)

Reply:
https://github.com/ansible/awx/blob/devel/awx/main/tasks/jobs.py#L531

https://github.com/ansible/awx/blob/devel/awx/main/models/credential.py#L553

https://github.com/ansible/awx_plugins.interfaces/blob/9af86ce7543377c63dabeb373159abc37bfa8ff4/src/awx_plugins/interfaces/_temporary_private_inject_api.py#L155

You see way down at the bottom it still seems to do:

```python
    if 'INVENTORY_UPDATE_ID' not in env:
        extra_vars = _build_extra_vars(
            sandbox_env,
            namespace,
            cred_type.injectors.get(
                'extra_vars', {},
            ),
        )
```

which is kind of terrible and klunky. But I guess it is at least expected to work.